app/channels/game_data_channel.rb

```diff
@@ -2,18 +2,66 @@ class GameDataChannel < ApplicationCable::Channel
   on_subscribe :welcome_player
 
   def subscribed
-    # ActionCable doesn't give access to params[] here if mounted in application.rb
-    @player = Player.includes(:game, :user).find_by(token: connection.current_player.token)
-    @player.update(subscribed: true)
-    stream_from "game_#{@player.game.game_key}"
+    current_player.update(subscribed: true)
+    stream_from current_player.game
     ensure_confirmation_sent
   end
 
   def unsubscribed
     # Any cleanup needed when channel is unsubscribed
   end
 
+  def send_hint(data)
+    # binding.pry
+    game = current_player.game
+    if game.current_player != current_player
+      illegal_action("#{current_player.name} attempted to submit a hint out of turn")
+    elsif !current_player.intel?
+      illegal_action("#{current_player.name} attempted to submit a hint, but doesn't have the Intel role")
+    elsif game.hint_invalid?(data[:hintWord])
+      illegal_action("#{current_player.name} attempted to submit an invalid hint")
+    else
+      game.advance!
+
+      hint = current_player.game.hints.create(
+        team: current_player.team,
+        word: data[:hintWord],
+        num: data[:numCards]
+      )
+
+      payload = {
+        type: 'hint-provided',
+        data: {
+          isBlueTeam: hint.blue?,
+          hintWord: hint.word,
+          relatedCards: hint.num
+        }
+      }
+
+      game.guesses_remaining = hint.num + 1
+      game.save
+      broadcast_message payload
+    end
+  end
+
   private
+
+          ##  ######   #######  ##    ##
+          ## ##    ## ##     ## ###   ##
```

Comment:
What's all this about ? Lol

Reply:
Since this class was getting a little large, I decided to add some ASCII section headings that are visible in the source map. :P